pyfaidx/__init__.py

```diff
@@ -303,14 +303,25 @@ def __init__(self, filename, default_seq=None, key_function=lambda x: x.split()[
             else:
                 self._fasta_opener = bgzf.open
                 self._bgzf = True
-        elif filenameLower.endswith('.gz') or filenameLower.endswith('.bz2'):
-            raise RuntimeError("Compressed FASTA is only supported in BGZF "
-                               "format, via the BioPython bgzf module.")
+        elif filenameLower.endswith('.bz2') or filenameLower.endswith('.zip'):
+            raise RuntimeError(
+                "Compressed FASTA is only supported in BGZF format. Use "
+                "bgzip to compresss your FASTA.")
         else:
             self._fasta_opener = open
             self._bgzf = False
 
-        self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        try:
+            self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        except ValueError as e:
+            if str(e).find('BGZF') > -1:
+                raise RuntimeError(
+                    "Compressed FASTA is only supported in BGZF format. Use "
+                    "the samtools bgzip utility (instead of gzip) to "
+                    "compresss your FASTA.")
+            else:
+                raise
```

Comment:
OK, I have simplified the Exception subclasses and changed line 55 in `cli.py` accordingly.

I removed `RegionError` as it appears to be unused.

I can stop with these little clean ups if you like. Else, I have another that I'd do: In `FastaRecord` and `MutableFastaRecord` you have a `try/except` that just re-raises the original exception. So the `try/except` isn't needed.

Sorry if I'm being a pain...